src/movetype.cpp

```diff
@@ -743,7 +743,12 @@ utils::string_map_res movetype::resistances::damage_table() const
  */
 int movetype::resistances::resistance_against(const attack_type & attack) const
 {
-	return cfg_[attack.type()].to_int(100);
+	std::pair<std::string, std::string> types = attack.damage_type();
+	int res = cfg_[types.first].to_int(100);
```

Comment:
`cfg_[types.first].to_int(100);` should be replaced with using the `resistance_against` method directly below this one.

Reply:
replaced